### PR TITLE
chore: Single UI Provider

### PR DESCRIPTION
**Pull request by tlgimenes**
## What's the purpose of this pull request?
This PR joins duplicated contexts we had on our code. 

## How does it work?
Both `UIProvider` and `ModalProvider` had a goal of controlling the global UI state of our site. This created an interdependency between these contexts, generating a mess on our code. For example, for openning the cart on UIProvider we relied on functions provided by `ModalProvider`. Also, when openning the cart we had to invoke a function called `onModalOpen`, what makes no sense since the cart is a drawer, not a modal.

This PR addresses this issue by merging UIProvider and ModalProvider into a single context so we don't have that mess anymore. Also, I have changed the name of `onModalClose` and `onModalOpen` to `fadeOut` and `fadeIn` respectively. 

After these changes, the minicart toggle button went from:
```
const { closeCart } = useUI()
const { onModalClose } = useModal()

const onClick = () => {
    onModalClose()
    closeCart()
}

``` 

to: 
```
const { closeCart, fadeOut } = useUI()

const onClick = () => {
    fadeOut()
    closeCart()
}
```

## How to test it?
Make sure all fade effects are working correctly, as well as the scroll behavior when the modal is open

### CODE REVIEW

**Review by danzanzini — CHANGES_REQUESTED** [review]
Requesting changes due to the scroll problem at the PDP. I've left one comment regarding this on code.

File: src/sdk/ui/modal/Provider.tsx [hunk 1]
```diff
@@ -1,81 +0,0 @@
...
-  const onModalClose = useCallback(() => {
-    setFade('out')
-    layout.current?.classList.remove('no-scroll')
-  }, [])
-
-  useEffect(() => {
-    layout.current = document.body
-
-    return () => {
-      layout.current?.classList.remove('no-scroll')
```
[the remark this answers is not in the document]
Comment (tlgimenes) [reply src/sdk/ui/modal/Provider.tsx:47]: nice catch! I'll take a look why

**Comment by tlgimenes** [comment]
Oh @danzanzini , I think it's working now :)

**Review by danzanzini — APPROVED** [review]
🚀

**Pull request merged**